File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/Ellipse/Ellipse.py
#=======================================================================
# 
# Bellhop: Ellipsoidal boundaries
# Mexilhoeira Grande, Dom Jul 30 11:53:49 WEST 2017
# Written by Tordar 
# 
#=======================================================================

from os import system
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from plotray import *

print("Bellhop - Ellipsoidal boundaries:") 
print("ray calculation including bathymetry")

fid = open('Ellipse.ati','r')
theline = fid.readline()
theline = fid.readline()
n       = int( theline )
ratikm  = zeros( n ) 
zati    = zeros( n )
for i in range(n):
    theline = str( fid.readline() )
    datai = theline.split()
    ratikm[ i] = float( datai[0] )
    zati[   i] = float( datai[1] )
fid.close()

rati = ratikm*1000

fid = open('Ellipse.bty','r')
theline = fid.readline()
theline = fid.readline()
n       = int( theline )
rbtykm  = zeros( n ) 
zbty    = zeros( n )
for i in range(n):
    theline = str( fid.readline() )
    datai = theline.split()
    rbtykm[ i] = float( datai[0] )
    zbty[   i] = float( datai[1] )
fid.close()

rbty = rbtykm*1000

system("bellhop.exe Ellipse")

figure(1)
plotray('Ellipse.ray')
plot(rati,-zati,'b',linewidth=2)
plot(rbty,-zbty,'k',linewidth=2)
xlabel('Range (m)')
ylabel('Depth (m)')
title("Bellhop - Ellipsoidal boundaries")

show()

print("done.")

File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/Ellipse/Ellipse.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/Plot
addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('Bellhop - Ellipsoidal boundaries:') 
disp('Ray trace run') 
disp('Geometric gaussian beams')

fid = fopen('Ellipse.ati','r');
s = fscanf(fid,'%s\n',[1 1]);
n = fscanf(fid,'%d\n',[1 1]);
data = fscanf(fid,'%f %f\n',[2 n]);
fclose(fid);

ratikm = data(1,:); rati = ratikm*1000; 
zati   = data(2,:);

fid = fopen('Ellipse.bty','r');
s = fscanf(fid,'%s\n',[1 1]);
n = fscanf(fid,'%d\n',[1 1]);
data = fscanf(fid,'%f %f\n',[2 n]);
fclose(fid);

rbtykm = data(1,:); rbty = rbtykm*1000; 
zbty   = data(2,:);

system('bellhop.exe Ellipse');

plotray('Ellipse.ray')
hold on
plot(rati,zati,'b','LineWidth',2)
plot(rbty,zbty,'k','LineWidth',2)
hold off

disp('done.')
